packages/backend/src/apps/formsg/common/webhook-settings.ts

```diff
@@ -0,0 +1,113 @@
+import { IBaseTrigger, IGlobalVariable } from '@plumber/types'
+
+import appConfig from '@/config/app'
+import logger from '@/helpers/logger'
+
+import { parseFormIdFormat } from '../auth/verify-credentials'
+
+export const FORMSG_WEBHOOK_VERIFICATION_MESSAGE = {
+  VERIFIED: 'Your form is connected successfully.',
+  EMPTY: 'Click "Connect" to connect your form to this pipe.',
+  ANOTHER_PIPE:
+    'The form you are trying to connect is currently being used in another pipe. Continuing with this connection will cause the other pipe to break.',
+  ANOTHER_ENDPOINT:
+    'The form is currently connected to a different endpoint. Continuing with this connection will override this setting.',
+  ERROR: "We couldn't verify your webhook settings. Please try again later.",
+}
+
+function getFormDetailsFromGlobalVariable($: IGlobalVariable) {
+  const userEmail = $.userEmail
+  if (!userEmail) {
+    throw new Error('Missing User Email')
+  }
+
+  const webhookUrl = $.webhookUrl
+  if (!webhookUrl) {
+    throw new Error('Webhook URL not instatiated')
+  }
+
+  const formId = parseFormIdFormat($)
+  if (!formId) {
+    throw new Error('Missing Form ID')
+  }
+
+  return {
+    userEmail,
+    webhookUrl,
+    formId,
+  }
+}
+
+export async function registerWebhookUrl(
+  $: IGlobalVariable,
+): ReturnType<IBaseTrigger['registerHook']> {
+  const { userEmail, webhookUrl, formId } = getFormDetailsFromGlobalVariable($)
+
+  try {
+    // Set formsg bearer token here
+    await $.http.patch(
+      `/public/v1/admin/forms/${formId}/settings`,
+      {
+        webhook: {
+          url: webhookUrl,
+          isRetryEnabled: true,
+        },
+        userEmail,
+      },
+      {
+        headers: {
+          Authorization: 'Bearer w' + appConfig.formsgApiKey,
+        },
+      },
+    )
+  } catch (e) {
+    logger.error('Unable to register webhook url', e)
```

Comment:
nit: can we haz more details in the logging in case we need to drill in logs later.  Do we also need entire error object - since I suspect stuff like response headers maybe not that important?

Maybe:
```
{
event: 'register-formsg-webhook-failed',
errorMessage: error.message,
errorResponse: error.response.data,
userEmail,
formId

}
```

Reply:
Good idea!